src/Microsoft.AspNetCore.Mvc.RazorPages/PageActionDescriptor.cs

```diff
@@ -2,15 +2,27 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System.Diagnostics;
+using System.Reflection;
 using Microsoft.AspNetCore.Mvc.Abstractions;
 
-namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
+namespace Microsoft.AspNetCore.Mvc.RazorPages
 {
     [DebuggerDisplay("{" + nameof(ViewEnginePath) + "}")]
     public class PageActionDescriptor : ActionDescriptor
     {
+        /// <summary>
+        /// Gets or sets the application root relative path for the page.
+        /// </summary>
         public string RelativePath { get; set; }
 
+        /// <summary>
+        /// Gets or sets the path relative to the base path for page discovery.
+        /// </summary>
         public string ViewEnginePath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="TypeInfo"/> of the page.
+        /// </summary>
+        public TypeInfo PageTypeInfo { get; set; }
```

Comment:
what happened to the CompiledPageAD? This is mixing things that require compilation with things that dont.